src/core/Akka.Remote/Endpoint.cs

```diff
@@ -320,44 +339,47 @@ internal class ReliableDeliverySupervisor : UntypedActor
         /// If we already have an inbound handle then UID is initially confirmed.
         /// (This actor is never restarted.)
         /// </summary>
-        public bool UidConfirmed { get; set; }
+        public bool UidConfirmed { get; private set; }
 
-        public Deadline BailoutAt
+        private Deadline _bailoutAt = null;
+
+        protected override SupervisorStrategy SupervisorStrategy()
         {
-            get { return Deadline.Now + _settings.InitialSysMsgDeliveryTimeout; }
+            return new OneForOneStrategy(ex =>
+            {
+                if (ex is IAssociationProblem)
+                    return Directive.Escalate;
+
+                _log.Warning("Association with remote system {0} has failed; address is now gated for {1} ms. Reason is: [{2}]", _remoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds, ex);
+                UidConfirmed = false; // Need confirmation of UID again
+                if (_bufferWasInUse)
```

Comment:
If we were ever buffering system messages, then we need to stop the `EndpointWriter` and let the `EndpointManger` know that we've stopped reading messages from the network (I'll talk about what that does in a moment, but essentially it completes a handoff process to a new set of EndpointReader and EndpointWriter actors.)

However, I think there might be an issue in the JVM implementation here potentially where this bit can be dirtied when connected to a previous instance of the node, and I filed that here: https://github.com/akka/akka/issues/19322